### DISCUSSION TITLE
Extracting data for 29302 SNP(s) from 1 GWAS(s) 1 of 1 outcomes  [>] 1 of 3 chunks Error in api_query("associations", query = list(variant = variants, id = id,  :    The query to MR-Base exceeded 300 seconds and timed out. Please simplify the query

### DISCUSSION THREAD

**Discussion by cnblack** [MRCIEU/TwoSampleMR#502] [General]
I've been trying to extract outcome data using extract_outcome_data(snps=exposure_dat$SNP, outcomes=outcomeID) from this package, but I keep getting an error: Error in api_query("associations", query = list(variant = variants, id = id, : The query to MR-Base exceeded 300 seconds and timed out. Please simplify the query. It's been like this for three days now. How can I fix it?

**Comment by Noralii** [comment]
I also met the same problem

**Comment by remlapmot** [comment]
That's alot of SNPs.

You need to update your TwoSampleMR and ieugwasr packages - you can use the following command to do this.

```r
install.packages("TwoSampleMR", repos = c("https://mrcieu.r-universe.dev", "https://cran.r-project.org"))
```

Then you need to obtain an OPENGWAS_JWT token - see the ieugwasr documentation https://mrcieu.github.io/ieugwasr/articles/guide.html - and store it in your .Renviron file - then restart R.

**Comment by DRleifeng** [comment]
So did I. May I ask how it has since been resolved?

**Comment by shiqi-shan** [comment]
Extracting data for 0 SNP(s) from 1 GWAS(s)
